src/components/ec-currency-input/ec-currency-input.vue

```diff
@@ -110,166 +110,151 @@
-      if (!content) {
-        return null;
-      }
-      return { content, placement };
-    },
-    currencyModel: {
-      get() {
-        return this.currenciesItems.find(item => item.value === this.modelValue.currency);
-      },
-      set(item) {
-        this.$emit('update:modelValue', { ...this.modelValue, currency: item.value });
-      },
-    },
-    amountModel: {
-      get() {
-        return this.modelValue.amount;
-      },
-      set(value) {
-        this.$emit('update:modelValue', { ...this.modelValue, amount: value });
-      },
-    },
+  currenciesAreLoading: {
+    type: Boolean,
+    default: false,
   },
-  methods: {
-    getPopoverStyle() {
-      if (this.$refs.popoverWidthReference) {
-        return {
-          width: `${this.$refs.popoverWidthReference.offsetWidth}px`,
-        };
-      }
-
-      return null;
-    },
-    onAmountChange(evt) {
-      this.$emit('change', evt);
-      this.$emit('amount-change', evt);
-    },
-    onCurrencyChange(evt) {
-      this.currenciesHasFocus = true;
-      this.$emit('change', evt);
-      this.$emit('currency-change', evt);
-    },
-    onFocusCurrency() {
-      this.currenciesHasFocus = true;
-      this.$emit('focus');
-    },
-    prefixedDataTest(dataTestSuffix) {
-      const dataTestPrefix = this.$attrs['data-test'];
-      if (dataTestPrefix) {
-        const dataTestPrefixes = [...dataTestPrefix.split(' '), 'ec-currency-input'];
-        return dataTestPrefixes.map(dataTest => `${dataTest}__${dataTestSuffix}`).join(' ');
-      }
-
-      return `ec-currency-input__${dataTestSuffix}`;
-    },
+  disabledCurrenciesTooltip: {
+    type: Object,
   },
+  isCurrenciesDisabled: {
+    type: Boolean,
+    default: false,
+  },
+  isAmountDisabled: {
+    type: Boolean,
+    default: false,
+  },
+  isSensitive: {
+    type: Boolean,
+    default: false,
+  },
+  amountPlaceholder: {
+    type: String,
+  },
+  searchCurrencyPlaceholder: {
+    type: String,
+    default: 'Search...',
+  },
+  noCurrenciesText: {
+    type: String,
+    default: 'No results found',
+  },
+});
+
+const emit = defineEmits(['update:modelValue', 'change', 'focus', 'amount-change', 'currency-change', 'open', 'after-open']);
+
+const uid = getUid();
+let currenciesHasFocus = false;
+const popoverOptions = {
+  autoSize: 'min',
 };
+const popoverWidthReference = ref(null);
+const attrs = useAttrs();
+
+const id = computed(() => `ec-currency-input-field-${uid}`);
+const errorId = computed(() => (isInvalid.value ? `ec-currency-input-field-${uid}` : null));
+const isInvalid = computed(() => !!props.errorMessage);
+const currenciesItems = computed(() => props.currencies.map(currency => ({ text: currency, value: currency, id: currency })));
+const currenciesTooltipOptions = computed(() => {
+  const { content, placement = 'top' } = props.disabledCurrenciesTooltip || {};
+  if (!content) {
+    return null;
+  }
+  return { content, placement };
+});
+const currencyModel = computed({
```

Comment:
```suggestion

const currencyModel = computed({
```

Code with blank line delimiters between big snippets or between logical different parts looks more readable.